### PR TITLE
Card feature updates

### PR DESCRIPTION
**Pull request by castastrophe**
Incorporate a few new features into the card layout:

- Support for background color in the header region
- Update variable names and add additional hooks
- Add overflow setting to allow children in slots to fill the card
- Add storybook support for self-closing tags (img, br, hr, etc.)
- Updates border-radius to 3px globally

### CODE REVIEW

**Comment by starryeyez024** [comment]
@castastrophe after running `npm run cleand && npm install`, the command `npm run storybook` returns an error at localhost:9001 that says: 
`Story of "Call to action" named "pfe-cta" already exists`

File: elements/pfe-card/src/pfe-card.js [hunk 1]
```diff
@@ -82,20 +82,30 @@ class PfeCard extends PFElement {
...
   get backgroundColor() {
-    return this.getAttribute("color") || "base";
+    return (
+      this.getAttribute("pfe-color") || this.getAttribute("color") || "base"
+    );
   }
 
   static get observedAttributes() {
-    return ["color"];
+    return ["pfe-color", "pfe-img-src", "pfe-size"];
```
Comment (chrisdo1) [inline elements/pfe-card/src/pfe-card.js]: @castastrophe I'm curious about the pfe-size=small attribute. Why not set this as a boolean? Would we want to leave it open for other size values later?

File: elements/pfe-card/src/pfe-card.scss [hunk 2]
```diff
@@ -1,127 +1,149 @@
...
+  // Start of style declarations for host element
+  display: flex;
+  flex-direction: column;
+  justify-items: flex-start;
 
-.pfe-card__body::slotted(*:nth-child(2)) { // child pseudo-selectors appear to be based on slotted code before upgrade.
-  margin-top: 0 !important;
-}
+  padding:       #{pfe-local(Padding)};
+  border:        #{pfe-local(Border)};
```
Comment (starryeyez024) [inline elements/pfe-card/src/pfe-card.scss]: I don't know that we want a border around the card by default... can the default color be transparent? It creates a rim around the card header:
screen shot 2019-02-19 at 4 43 12 pm

**Comment by starryeyez024** [comment]
http://pastebin.test.redhat.com/718499

File: elements/pfe-card/src/pfe-card.scss (hunk 2 above)
Comment (castastrophe) [reply elements/pfe-card/src/pfe-card.scss]: The border color is set to transparent already but the fact that it has width is pushing the header in.  I think I need to set the weight to 0px instead to fix this.

File: elements/pfe-card/src/pfe-card.js (hunk 1 above)
Comment (castastrophe) [reply elements/pfe-card/src/pfe-card.js]: Yes and I wanted to match the approach we've taken in other components too

File: elements/pfe-card/src/pfe-card.scss (hunk 2 above)
Comment (castastrophe) [reply elements/pfe-card/src/pfe-card.scss]: Fixed this.

**Comment by kylebuch8** [comment]
Alright. I think we're finally good to go with card. Just a heads up, we're making a few changes to the PFElement base class.

In the `connectedCallback()`, we're adding `this.setAttribute("pfelement", "")` and then in the `_mapSchemaToSlots` method, we're using `this.setAttribute(`has_${slot}`, "")` if the slot exists.

This is all done because set classes might be fragile.

[View this comment](https://github.com/angular/angular/issues/15399#issuecomment-318785677) to see what helped guide us to these changes.

**Pull request merged**